src/lib/controllers/configuration-manager/generators/generate-public-key.ts

```diff
@@ -0,0 +1,14 @@
+import * as forge from 'node-forge';
+
+/**
+ * GENERATE_PUBLIC_KEY
+ * @param {forge.pki.PEM} privateKeyPEM
+ * @returns {Promise<string>} PublicKeyPEM
+ */
+export async function generatePublicKey(privateKeyPEM: forge.pki.PEM) {
+	const privateKey = forge.pki.privateKeyFromPem(
+		privateKeyPEM,
+	) as forge.pki.rsa.PrivateKey;
```

Comment:
Not already conforming to type?

Reply:
Again, we need to restrict to the rsa.PrivateKey type, as the return typ is defined as: `type PrivateKey = rsa.PrivateKey | ed25519.Key;`